tools/mpremote/tests/test_filesystem.sh

```diff
@@ -170,3 +170,20 @@ EOF
 $MPREMOTE resume cp -r "${TMP}/package" :
 $MPREMOTE resume ls : :package :package/subpackage
 $MPREMOTE resume exec "import package; package.x(); package.y()"
+
+# Test rm -r functionality
+echo -----
+$MPREMOTE resume mkdir :testdir
+$MPREMOTE resume cp -r "${TMP}/package" :testdir/package
```

Comment:
Aside from the non-empty diff with `test_filesystem.sh.exp`, this line also generates another exception, such that the remaining test lines aren't even run:

```
~/micropython/tools/mpremote/tests$ ./run-mpremote-tests.sh 
./test_eval_exec_run.sh: OK
./test_filesystem.sh: FAIL
194,201d193
< -----
< mkdir :testdir
< cp ${TMP}/package :testdir/package
< mpremote: Error with transport:
< Traceback (most recent call last):
<   File "<stdin>", line 1, in <module>
< OSError: 28
< 
./test_mip_local_install.sh: OK
./test_mount.sh: OK
./test_recursive_cp.sh: OK
./test_resume.sh: OK
```

Reply:
Thanks for the report ,  Ill  dig-in to the test setup on linux

Reply:
> Thanks for the report , Ill dig-in to the test setup on linux

If it helps, OSError 28 is ENOSPC, No space left on device.

Reply:
Ok, yes, I can confirm, increasing the size of the filesystem from 50 blocks to 100 blocks fixes this first issue.

Reply:
As written, in the state that the preceding tests leave the ramdisk filesystem, there is not enough space on that ramdisk for this test to pass.

Adjusting the number of blocks in `ramdisk.py` at the start of the test from 50 to 100 fixes this issue, at least.

Please run the entire test script end-to-end yourself, on a _clean_ repository -- no uncommitted files or changes; as if you'd just run `git reset --hard`.

Reply:
the problem with one big sequential test is that not all tests clean up at teardown.
Avoided the `OSError 28 is ENOSPC` issue by restarting the ramdrive in order to be able to start in a clean known state.